### PR TITLE
Ukraine

### PR DESCRIPTION
**Pull request by ciscorucinski**
## Summary
Adding Ukraine scraper

## Changes
- Added scraper
- Added me as a maintainer

## Additional notes
None

### CODE REVIEW

File: src/shared/scrapers/UKR/index.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  'Kharkiv region': 'iso2:UA-63',
+  'Kherson region': 'iso2:UA-65',
+
+  'Khmelnytskyi region': 'iso2:UA-68',
+  'Khmelnytsky region': 'iso2:UA-68',
+
+  'Kirovohrad region': 'iso2:UA-35',
+  'Kirovograd region': 'iso2:UA-35',
+
+  Kyiv: 'iso2:UA-30',
```
Comment (hyperknot) [inline src/shared/scrapers/UKR/index.js:24]: Kyiv would never be used, would it?

File: src/shared/scrapers/UKR/index.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+  'Lviv region': 'iso2:UA-46',
+  'Mykolaiv region': 'iso2:UA-48',
+
+  'Odesa region': 'iso2:UA-51',
+  'Odessa region': 'iso2:UA-51',
+
+  'Poltava region': 'iso2:UA-53',
+  'Rivne region': 'iso2:UA-56',
+  Sevastopol: 'iso2:UA-40',
```
Comment (hyperknot) [inline src/shared/scrapers/UKR/index.js:41]: Also, isn't it "Sevastopol region"?

**Comment by hyperknot** [comment]
Thanks for this, it looks amazing! 

1. Can you rename the folder to UA?

**Comment by hyperknot** [comment]
image

File: src/shared/scrapers/UKR/index.js (hunk 1 above)
Comment (ciscorucinski) [reply src/shared/scrapers/UKR/index.js:24]: For all the duplicate values, the top one comes from the list you gave me before, the second one comes from what is in the JSON

Comment (hyperknot) [reply src/shared/scrapers/UKR/index.js]: The list I gave is not important, only the ids on it. We just need to map to the ids. You can even do `.replace(/region/g, '')`and just make everything shorter.

File: src/shared/scrapers/UKR/index.js (hunk 2 above)
Comment (ciscorucinski) [reply src/shared/scrapers/UKR/index.js]: looking into this. I think some regions aren't included. I thought there was a mismatch between the list you gave me and what the JSON had.

Comment (ciscorucinski) [reply src/shared/scrapers/UKR/index.js]: `iso2:UA:40` is a city https://en.wikipedia.org/wiki/ISO_3166-2:UA

**Comment by hyperknot** [comment]
So actually I'd recommend:
1. Removing all the following logic:
```
region.label.en === 'Kyiv'
            ? geography.addCounty(region.label.en, 'city')
            : geography.addCounty(region.label.en, 'region');
```
2. Using the exact `region.label.en` strings in the mapping.
3. Adding a console.error() + continue when a region appears which is not in the map.

**Comment by ciscorucinski** [comment]
I don't need to have the `geography.addCounty(region.label.en, 'region');` code at least in one place?

**Comment by hyperknot** [comment]
I'll test it out if I can get it done without the geography.addCounty bits.

**Comment by hyperknot** [comment]
@ciscorucinski I pushed a commit to the PR. It explains what I meant by simplifying the code. Names really don't matter here, only the ids have to be correctly identified. If you are ok with the changes I can merge this tomorrow.

**Comment by hyperknot** [comment]
image

**Comment by lazd** [comment]
This looks great! Is the Ukraine source a timeseries? If so, there is a special way to handle it.

**Comment by lazd** [comment]
Gonna merge this as-is because it's awesome, but let's try to figure out if their api has a `from` field where we can get a full timeseries.

**Comment by lazd** [comment]
Gonna merge this as-is because it's awesome, but let's try to figure out if their api has a `from` field where we can get a full timeseries.

I'll make a few changes to the way `this.url` is used and fix caching (should not pass `false` as second argument), but this is gold. Thank you!

**Comment by lazd** [comment]
I now see why you pass `false` -- you can't go back in time for the timeseries otherwise, of course. But the URL bit needed to not modify `this.url` with the date, or we append the date for each date in the timeseries as we run (not true since we moved to separate processes, but we want to work under the assumption that the module is not re-imported for each day).

**Comment by shaperilio** [comment]
> I now see why you pass `false` -- you can't go back in time for the timeseries otherwise, of course. But the URL bit needed to not modify `this.url` with the date, or we append the date for each date in the timeseries as we run (not true since we moved to separate processes, but we want to work under the assumption that the module is not re-imported for each day).

You're referring to the `false` in the `const data = await fetch.json(this.url, false, { disableSSL: true })` call? Agreed, we don't want this because it means it never uses the cache - this is what I meant by having a need to "back-fill" cache. This source is a good example of that; [Argentina](#633) also has previous data up and would also need this type of functionality.

**Comment by shaperilio** [comment]
I just tested this, it *does* use the cache, and it can "go back in time", so disregard what I wrote.

**Pull request merged**